api_app/services/aad_authentication.py

```diff
@@ -236,11 +238,9 @@ def get_workspace_role_assignment_details(self, workspace: Workspace):
         workspace_role_assignments_details = defaultdict(list)
         for role_assignment in roles_graph_data["value"]:
             if role_assignment["principalType"] == "User" and role_assignment["principalId"] in user_emails:
-                if role_assignment["appRoleId"] == researcher_app_role_id:
-                    workspace_role_assignments_details["researcher_emails"].append(user_emails[role_assignment["principalId"]])
-                elif role_assignment["appRoleId"] == owner_app_role_id:
-                    workspace_role_assignments_details["owner_emails"].append(user_emails[role_assignment["principalId"]])
-
+                if role_assignment["appRoleId"] in app_role_ids.values():
```

Comment:
replace the if with and, maybe extract into a boolean or two for readability

Reply:
Fixed